### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by s3nnou**

### CODE REVIEW

File: src/UnitTesting/KataSum.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
+﻿using System.Text.RegularExpressions;
+
+namespace UnitTesting
+{
+    public interface IKataSum
+    {
+        public int Sum(string num1, string num2);
+
+    }
+    public class KataSum : IKataSum
```
Comment (DaryaKrauchukTR) [inline src/UnitTesting/KataSum.cs:10]: I think it is wrong to throw out exceptions. If the string is not a number, it should be 0 (zero) according to the task.
"4", "5" --> "9"
"34", "5" --> "39"
"", "" --> "0"
"2", "" --> "2"
"-5", "3" --> "-2"

Comment (Darya713) [reply src/UnitTesting/KataSum.cs:10]: "-5", "3" --> "3"

File: src/UnitTesting/FizzBuzz.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                fizzBuzzBuilder.Append("Fizz");
+            }
+
+            if (IsDivededByFive(number))
+            {
+                fizzBuzzBuilder.Append("Buzz");
+            }
+
+            var fizzBuzzString = fizzBuzzBuilder.ToString();
+            return string.IsNullOrEmpty(fizzBuzzString) ? number.ToString() : fizzBuzzString;
```
Comment (DaryaKrauchukTR) [inline src/UnitTesting/FizzBuzz.cs:49]: Use the string builder to check for null or empty to reduce the number of allocations.

File: src/UnitTesting/RecentlyUsedList.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        public RecentlyUsedList(int limit)
+        {
+            _items = new List<string>();
+            Count = 0;
+            Size = limit;
+        }
+
+        public int Count { get; private set; }
+
+        public int Size { get; private set; }
```
Comment (DaryaKrauchukTR) [inline src/UnitTesting/RecentlyUsedList.cs:32]: According to the task, the size of the list can be either with restriction ( there are two options - specified or default), or without restriction

File: src/UnitTesting/RecentlyUsedList.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        }
+
+        public RecentlyUsedList(int limit)
+        {
+            _items = new List<string>();
+            Count = 0;
+            Size = limit;
+        }
+
+        public int Count { get; private set; }
```
Comment (DaryaKrauchukTR) [inline src/UnitTesting/RecentlyUsedList.cs:30]: Add public properties, methods and indexer to the interface

File: test/UnitTests.Tests/RecentlyUsedListTests.cs [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+            list.Count.Should().Be(expectedSize);
+        }
+
+        [Test]
+        public void Add_WhenDuplicatesExistsInList_ShouldThrowExceptionAndAbortAdding()
+        {
+            // Arrange
+            var list = new RecentlyUsedList();
+            var firstItem = "hello";
+            var secondItem = "hello";
```
Comment (DaryaKrauchukTR) [inline test/UnitTests.Tests/RecentlyUsedListTests.cs:76]: Please use the different values

File: test/UnitTests.Tests/RecentlyUsedListTests.cs [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+            // Act
+            var list = new RecentlyUsedList(size);
+
+            // Assert
+            list.Size.Should().Be(expectedSize);
+        }
+
+        [Test]
+        public void Add_WhenCountIsBiggerThenSizw_ShouldRemoveNewItems()
```
Comment (DaryaKrauchukTR) [inline test/UnitTests.Tests/RecentlyUsedListTests.cs:131]: Sizw is a typo.

File: src/UnitTesting/FizzBuzz.cs [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        string PrintAll();
+    }
+
+    public class FizzBuzz : IFizzBuzz
+    {
+        public string PrintAll()
+        {
+            var stringBuilder = new StringBuilder();
+
+            for (var i = 1; i <= 100; i++)
```
Comment (DaryaKrauchukTR) [inline src/UnitTesting/FizzBuzz.cs:18]: magic numbers

File: src/UnitTesting/FizzBuzz.cs [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+            {
+                stringBuilder.AppendLine(PrintNumber(i));
+            }
+
+            return stringBuilder.ToString();
+        }
+
+        public string PrintNumber(int number)
+        {
+            if (number <= 0 || number > 100)
```
Comment (DaryaKrauchukTR) [inline src/UnitTesting/FizzBuzz.cs:28]: magic numbers

File: src/UnitTesting/KataSum.cs [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+                for (int i = 0; i < stringValue.Length; i++)
+                {
+                    y = y * 10 + (stringValue[i] - '0');
+                }
+
+            }
+
+            return y;
+        }
+
```
Comment (DaryaKrauchukTR) [inline src/UnitTesting/KataSum.cs:52]: blank line

**Review by DaryaKrauchukTR — APPROVED** [review]

**Pull request merged**